File: 0162-find-peak-element/0162-find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        begin = 0
        end = len(nums)
        while begin < end:
            middle = (begin + end) // 2
            if (0 == middle or nums[middle - 1] < nums[middle]) and (len(nums) == middle + 1 or nums[middle + 1] < nums[middle]):
                return middle

            if 0 < middle and nums[middle - 1] >= nums[middle]:
                end = middle
            else:
                begin = middle + 1

File: 0162-find-peak-element/0162-find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {
        int i;
        for (i = 0; i < nums.length; i++) {
            if ((0 == i || nums[i - 1] < nums[i])
                && (nums.length - 1 == i || nums[i] > nums[i + 1])) {
                break;
            }
        }

        return i;
    }
}